sormas-backend/src/main/java/de/symeda/sormas/backend/feature/FeatureConfigurationFacadeEjb.java

```diff
@@ -98,14 +121,116 @@
+		FeatureConfigurationCriteria criteria,
+		Integer offset,
+		Integer size,
+		List<SortProperty> sortProperties) {
+
+		if (criteria == null || criteria.getDisease() == null) {
+			throw new IllegalArgumentException("disease field cannot be null!");
+		}
+
+		CountryReferenceDto serverCountry = countryFacadeEjb.getServerCountry();
+		CriteriaBuilder cb = em.getCriteriaBuilder();
+		CriteriaQuery<FeatureConfigurationIndexDto> cq = cb.createQuery(FeatureConfigurationIndexDto.class);
+		Root<District> root = cq.from(District.class);
+		Join<District, Region> regionJoin = root.join(District.REGION, JoinType.LEFT);
+		Join<District, FeatureConfiguration> featureConfigurationJoin = root.join(District.FEATURE_CONFIGURATIONS, JoinType.LEFT);
+		Predicate filterJoinOnDisease = CriteriaBuilderHelper.or(
+			cb,
+			cb.and(cb.equal(featureConfigurationJoin.get(FeatureConfiguration.DISEASE), criteria.getDisease())),
+			cb.and(cb.isNull(featureConfigurationJoin.get(FeatureConfiguration.DISEASE))));
+		featureConfigurationJoin.on(filterJoinOnDisease);
+
+		cq.multiselect(
+			featureConfigurationJoin.get(FeatureConfiguration.UUID),
+			regionJoin.get(Region.UUID),
+			regionJoin.get(Region.NAME),
+			root.get(District.UUID),
+			root.get(District.NAME),
+			featureConfigurationJoin.get(FeatureConfiguration.DISEASE),
+			featureConfigurationJoin.get(FeatureConfiguration.ENABLED),
+			featureConfigurationJoin.get(FeatureConfiguration.END_DATE));
+
+		Predicate filter = null;
+
+		if (serverCountry != null) {
+			Path<Object> countryUuid = regionJoin.join(Region.COUNTRY, JoinType.LEFT).get(Country.UUID);
+			filter = CriteriaBuilderHelper.and(cb, filter, cb.or(cb.isNull(countryUuid), cb.equal(countryUuid, serverCountry.getUuid())));
+		}
+
+		if (criteria.getRegion() != null) {
+			filter = CriteriaBuilderHelper.and(cb, filter, cb.equal(regionJoin.get(Region.UUID), criteria.getRegion().getUuid()));
+
+		}
+		if (criteria.getDistrict() != null) {
+			filter = CriteriaBuilderHelper.and(cb, filter, cb.equal(root.get(District.UUID), criteria.getDistrict().getUuid()));
+		}
+
+		if (filter != null) {
+			cq.where(filter);
+		}
+
+		if (CollectionUtils.isNotEmpty(sortProperties)) {
+			List<Order> order = new ArrayList<>(sortProperties.size());
+			for (SortProperty sortProperty : sortProperties) {
+				Expression<?> expression;
+				switch (sortProperty.propertyName) {
+				case FeatureConfiguration.REGION:
+					expression = regionJoin.get(Region.NAME);
+					break;
+				case FeatureConfiguration.DISTRICT:
+					expression = root.get(District.NAME);
+					break;
+				default:
+					throw new IllegalArgumentException(sortProperty.propertyName);
+				}
+				order.add(sortProperty.ascending ? cb.asc(expression) : cb.desc(expression));
+			}
+			cq.orderBy(order);
+		} else {
+			cq.orderBy(cb.asc(regionJoin.get(Region.NAME)));
+		}
+		return QueryHelper.getResultList(em, cq, offset, size);
+
+	}
+
+	public Long count(DistrictCriteria criteria) {
```

Comment:
Remove this method and instead directly call the count method of DistrictFacadeEjb where it's used